src/charm.py

```diff
@@ -151,21 +137,34 @@ def _process_wheel_file(self, event):
 
                 # Push wheel file to the container and extract it.
                 container.push("/user_provided/wheel_file.whl", wheel_data, make_dirs=True)
```

Comment:
maybe you would want to remove the hardcoded wheel_file.whl from here and get it from resource_path

Reply:
Noted, will make this change in the next PR along with some minor changes.